### PR TITLE
fix: compilation of native modules on windows with older msvc versions

### PR DESCRIPTION
**Pull request by deepak1556**
#### Description of Change

https://chromium-review.googlesource.com/c/v8/v8/+/1847361 started using C++ `[[deprecated]]` attribute which has issues when building with msvc from VS2015. Native modules compiling with this version will have error of the format ` error C2416: attribute 'deprecated("..")' cannot be applied in this context` from v8 headers

This can be verified with the following test case:

```cpp
class A {
  public:
    [[deprecated("test")]]
    A(int a);
};

int main() {
  return 0;
}
```

with 2015 toolchain there will be error:

```ps
>  & 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\cl.exe' /c /I"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include" test.cc
```

with 2017 toolchain it will compile:

```ps
> & 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x64\cl.exe' /c /I"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include" test.cc
```

There is a msvc bug report out there which fixed this specific case, don't have the link. The PR just uses `__declspec(deprecated)` for those versions.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: fix compilation error for native modules building with VS 2015

### CODE REVIEW

File: patches/v8/fix_build_deprecated_attirbute_for_older_msvc_versions.patch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
++++ b/include/v8config.h
+@@ -388,10 +388,13 @@
+ # define V8_NOINLINE /* NOT SUPPORTED */
+ #endif
+ 
+-
+ // A macro (V8_DEPRECATED) to mark classes or functions as deprecated.
+ #if defined(V8_DEPRECATION_WARNINGS)
+-# define V8_DEPRECATED(message) [[deprecated(message)]]
++# if defined(_MSC_VER) && _MSC_VER <= 1900
```
Comment (deepak1556) [inline patches/v8/fix_build_deprecated_attirbute_for_older_msvc_versions.patch]: The version number is extracted from https://devblogs.microsoft.com/cppblog/side-by-side-minor-version-msvc-toolsets-in-visual-studio-2017/

**Comment by deepak1556** [comment]
Yeah its better to upstream this, will do it tomorrow.

**Comment by MarshallOfSound** [comment]
@deepak1556 Can you take this back to 8?

**Comment by deepak1556** [comment]
Upstream CL https://chromium-review.googlesource.com/c/v8/v8/+/2032196

**Comment by deepak1556** [comment]
Please see the above CL and the v8 team has responded with their supported release lines wrt msvc, so its better we remove this patch for Electron 9 and above. Let users fix this properly on their end using VS2017 build tools.

**Pull request merged**